app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Enums\Status;
+use App\Http\Requests\Enum;
+use App\Models\User;
+use App\Models\Authentication;
+use App\Http\Requests\UserRequest;
+use Carbon\Carbon;
+use Cookie;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Foundation\Auth\RegistersUsers;
+use Illuminate\Auth\Events\Registered;
+use App\Http\Controllers\Controller;
+
+class UserController extends Controller
+{
+    public function create(Request $request)
+    {
+        $enum = $request->enum;
+        $token = $request->token;  
+        $authentication = Authentication::where('token', $token)->orderBy('created_at', 'DESC')->orderBy('id', 'DESC')->first();
+
+        $current_time = now();
+        $token_expiry = $authentication->expired_at;  
+
+        if ($current_time > $token_expiry) {
+            return view('users.token_expired_error');
+        }
+
+        if ($authentication && $authentication->status === Status::SEND_MAIL) {
+            $authentication->status = Status::MAIL_VERIFY;
+            $authentication->save();
+            } else {
+                return view('users.invalid_token_error');
+            }
+
+        Cookie::queue('token', $token);
+        return view('users.create');  
+    }
+
+    public function store(UserRequest $request)
+    {
+        $token = $request->cookie('token');
+        $authentication = Authentication::where('token',$token)->first();
```

Comment:
```suggestion
        $authentication = Authentication::where('token', $token)->first();
```
半角スペース開けましょう！
と、完了したトークンも取得できちゃってるから重複登録できてしまいますね。

Reply:
承知しました。
重複登録できないよう変更しました。